sp/src/game/client/c_baselesson.cpp

```diff
@@ -666,7 +666,8 @@ void CIconLesson::UpdateInactive()
 			CUtlBuffer msg_data;
 			msg_data.PutChar( 1 );
 			msg_data.PutString( m_szHudHint.String() );
-			usermessages->DispatchUserMessage( usermessages->LookupUserMessage( "KeyHintText" ), bf_read( msg_data.Base(), msg_data.TellPut() ) );
```

Comment:
The issue here is that `DispatchUserMessage()` expects a (non-`const`) `bf_read&`. rvalues (unnamed temporaries) are normally not allowed to be used as non-`const` references (`const` would be fine). MSVC seems to allow it anyways. By using an explicit variable, it is no longer an rvalue and thus can be passed as non-`const` reference.

While looking into this I read the implementation of `DispatchUserMessage()`. For every hook it copies the `bf_read`, which seems unnecessary to me, considering there's a `Reset()` function on it. So if you wanted to you can probably optimize it to avoid making the copies and instead reset it before invoking each hook.